chain/chain/src/flat_storage_creator.rs

```diff
@@ -232,9 +202,9 @@ impl FlatStorageShardCreator {
                     let store = self.runtime_adapter.store().clone();
                     let epoch_id = self.runtime_adapter.get_epoch_id(&block_hash)?;
                     let shard_uid = self.runtime_adapter.shard_id_to_uid(shard_id, &epoch_id)?;
-                    let trie_storage = TrieDBStorage::new(store.clone(), shard_uid);
+                    let trie_storage = TrieDBStorage::new(store, shard_uid);
                     let state_root =
-                        chain_store.get_chunk_extra(&block_hash, &shard_uid)?.state_root().clone();
+                        *chain_store.get_chunk_extra(&block_hash, &shard_uid)?.state_root();
```

Comment:
Out of curiosity, why do you prefer dereferencing over clone? It looks right but I didn't find any obvious proofs that it is better.

Reply:
this is required to make clippy happy, it was introduced in https://github.com/near/nearcore/pull/8569

the error in this particular case:
```
error: using `clone` on type `near_primitives::hash::CryptoHash` which implements the `Copy` trait
...                                                                                                                                                             
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy                                                    
    = note: requested on the command line with `-D clippy::clone-on-copy`  
 ```